pkg/autoupdate/findnew.go

```diff
@@ -266,6 +263,41 @@ func getUpdateDir(fullBinaryPath string) string {
 	return fmt.Sprintf("%s%s", components[0], updateDirSuffix)
 }
 
+// Find the possible updates. filepath.Glob returns a list of things
+// that match the requested pattern. We sort the list to ensure that
+// we can tell which ones are earlier or later (remember, these are
+// timestamps).
+func getPossibleUpdates(updateDir, binaryName string) ([]string, error) {
+	// If this is launcher running on macOS, then we should have app bundles available instead --
+	// check for those first.
+	if runtime.GOOS == "darwin" {
+		binarySuffix := filepath.Join("Contents", "MacOS", binaryName)
+		fileGlob := filepath.Join(updateDir, "*", "*.app", binarySuffix)
+		possibleUpdates, err := filepath.Glob(fileGlob)
+
+		if err == nil && len(possibleUpdates) > 0 {
+			appBundleNames := make([]string, len(possibleUpdates))
+			for i, binaryPath := range possibleUpdates {
+				appBundleNames[i] = strings.TrimSuffix(binaryPath, "/"+binarySuffix)
+			}
+			sort.Strings(appBundleNames)
+			return appBundleNames, nil
+		}
```

Comment:
I feel like we should return the error here if we get it back, looking at the docs

> Glob returns the names of all files matching pattern or nil if there is no matching file. The syntax of patterns is the same as in Match. The pattern may describe hierarchical names such as /usr/*/bin/ed (assuming the Separator is '/').
> 
> Glob ignores file system errors such as I/O errors reading directories. The only possible returned error is ErrBadPattern, when pattern is malformed.


```suggestion
		possibleUpdates, err := filepath.Glob(fileGlob)
		if err != nil {
		  return nil, fmt.Errorf("globbing updates: %w, err")
		}

		if len(possibleUpdates) > 0 {
			appBundleNames := make([]string, len(possibleUpdates))
			for i, binaryPath := range possibleUpdates {
				appBundleNames[i] = strings.TrimSuffix(binaryPath, "/"+binarySuffix)
			}
			sort.Strings(appBundleNames)
			return appBundleNames, nil
		}
```

Reply:
I thought it would be safer to not return the error because then we can fall back (below) to the previous method of looking for possible updates. Wouldn't we want this in the case where the pattern _is_ malformed, so that eventually it could update itself to a newer version where the pattern is fixed? (If yes I should probably update to add a log for the error anyway, though.)

What you suggested is more in line with previous behavior on globbing errors, though, so I'm happy to do it that way too.

Reply:
Good point! I agree with logging the error and the going to the fall back, especially while this is new.

Reply:
Maybe comment? Something like `We can ignore this error -- we fall back to the prior behavior below`

Reply:
Added a log message and a comment 🙂